packages/client/components/TaskEditor/decorators.ts

```diff
@@ -12,11 +13,34 @@ const findEntity = (entityType) => (contentBlock, callback, contentState) => {
   }, callback)
 }
 
+const findText = (searchQuery) => (contentBlock, callback) => {
+  if (searchQuery == null || searchQuery === '') {
+    return
+  }
+
+  const text = contentBlock.getText()
+  const textLower = text.toLowerCase()
+  const searchQueryLower = searchQuery.toLowerCase()
+
+  let start = 0
+  let found = true
+  while (found) {
```

Comment:
This is probably just personal preference, but I find `find` methods to be more readable than `while` loops. Would something like this work?
```
textLower.split('').findIndex((letter) => searchQueryLower.includes(letter))
```

Reply:
@nickoferrall Here, I need to find all occurrences `startindex` and `endIndex` without overlapping and call a callback function for each found.

Not sure I got the idea with `split` and `findIndex`. 
With `split` I see kind of following solution:

```js
textLower.split(searchQueryLower).map(function (token) {
    return this.pos += token.length + searchQueryLower.length
}, { pos: -searchQueryLower.length }).slice(0, -1).forEach(i => { 
  callback(i, i + searchQueryLower.length)
})
```

What do you think?

Reply:
Ahh OK, thanks for clarifying! If we can avoid making `findText` a thunk, I would be very happy as I find them quite tricky. That might just be me so feel free to ignore this suggestion and stick with your original version, but could this work?

```
  if (searchQuery) {
    const findMatchingText = (
      contentBlock: ContentBlock,
      callback: (startIdx: number, endIdx: number) => void
    ) => {
      const textLower = contentBlock.getText().toLowerCase()
      const searchQueryLower = searchQuery.toLowerCase()
      const matchingIdx = textLower.indexOf(searchQueryLower)
      if (matchingIdx !== -1) {
        callback(matchingIdx, matchingIdx + searchQueryLower.length)
      }
    }

    compositeDecorator.push({
      strategy: findMatchingText,
      component: SearchHighlight
    })
  }
```

Reply:
@nickoferrall yeah, this removes thunk, but iteration is still needed. Made it as you suggested.

Reply:
@igorlesnenko why do we need the while loop as well as `indexOf`? `indexOf` is already iterating through the array?

Reply:
Ahh OK cool. Thanks, that makes sense. LGTM!